### ISSUE TITLE
Cannot install TexLive 2023

### ISSUE THREAD

**Issue by eric-wieser** [teatimeguest/setup-texlive-action#304]
Running with
```yaml
      - uses: teatimeguest/setup-texlive-action@v3
        with:
          package-file: |
            .github/tl_packages
          version: "2023"
```
gives
```
Downloading install-tl-unx.tar.gz from https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2023/tlnet-final/install-tl-unx.tar.gz
  unable to verify the first certificate
  Waiting 17 seconds before trying again
  unable to verify the first certificate
  Waiting 17 seconds before trying again
  | InstallTLError: Failed to download install-tl
  |     at download (teatimeguest/setup-texlive-action/v3/dist/index.mjs:87795:19)
  |     at async acquire (teatimeguest/setup-texlive-action/v3/dist/index.mjs:87755:36)
  |     at async install2 (teatimeguest/setup-texlive-action/v3/dist/index.mjs:88844:21)
  |     at async teatimeguest/setup-texlive-action/v3/dist/index.mjs:88994:9 {
  |   code: 'FAILED_TO_DOWNLOAD',
  |   version: '2023',
  |   repository: 'https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2023/tlnet-final/',
  |   [cause]: Error: unable to verify the first certificate
  |    {
  |     code: 'UNABLE_TO_VERIFY_LEAF_SIGNATURE'
  |   }
  | }
  InstallTLError: Failed to download install-tl
| InstallTLError: Download from FTP repositories is currently not supported
|     at download (teatimeguest/setup-texlive-action/v3/dist/index.mjs:87782:11)
|     at acquire (teatimeguest/setup-texlive-action/v3/dist/index.mjs:87755:42)
|     at install2 (teatimeguest/setup-texlive-action/v3/dist/index.mjs:88844:[27](https://github.com/eric-wieser/thesis/actions/runs/9529813813/job/26268956178#step:4:30))
|     at async teatimeguest/setup-texlive-action/v3/dist/index.mjs:88994:9 {
|   code: 'FAILED_TO_DOWNLOAD',
|   version: '2023',
|   repository: 'ftp://tug.org/historic/systems/texlive/2023/tlnet-final/'
| }
Error: InstallTLError: Download from FTP repositories is currently not supported
```
from
https://github.com/teatimeguest/setup-texlive-action/blob/d1e14166b5bc33b88117cdd9c446a03f29be242f/packages/texlive/src/install-tl/cli.ts#L137-L142

**Comment by teatimeguest** [comment]
Thank you for reporting the issue!

There seems to be a certification problem with `https://ftp.math.utah.edu/pub/tex/historic/`:
> ```console
> Downloading install-tl-unx.tar.gz from https://ftp.math.utah.edu/pub/tex/historic/systems/texlive/2023/tlnet-final/install-tl-unx.tar.gz
>   unable to verify the first certificate
> ```
In such cases, a workaround is to specify a repository explicitly:
```yaml
- uses: teatimeguest/setup-texlive-action@v3
  with:
    package-file: |
      .github/tl_packages
    repository: https://mirrors.tuna.tsinghua.edu.cn/tex-historic-archive/systems/texlive/2023/tlnet-final/
    version: 2023
```
Available servers are listed at <https://tug.org/historic/>.

**Comment by teatimeguest** [comment]
@eric-wieser I have released [`v3.3.0`](https://github.com/teatimeguest/setup-texlive-action/releases/tag/v3.3.0) with a workaround for this problem.

If you still have problems, please feel free to reopen the issue.
